File: src/main/java/org/aikidodanang/service/PostServiceImpl.java
package org.aikidodanang.service;

import org.aikidodanang.model.Post;
import org.aikidodanang.repository.PostRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@Transactional
public class PostServiceImpl implements PostService {
    private final PostRepository postRepository;

    public PostServiceImpl(PostRepository postRepository) {
        this.postRepository = postRepository;
    }

    @Override
    public Post findByTitle(String title) {
        return postRepository.findByTitle(title);
    }

    @Override
    public List<Post> findByArticle(boolean isArticle) {
        return postRepository.findByArticle(isArticle);
    }
}

File: src/main/kotlin/org/aikidodanang/services/PostServiceImpl.kt
package org.aikidodanang.services

import org.aikidodanang.dto.PostDto
import org.aikidodanang.repository.PostRepository
import org.springframework.stereotype.Service

@Service
class PostServiceImpl(
        val postRepository: PostRepository
) : PostService {
    override fun findByTitle(title: String): PostDto {
        return PostDto.fromPost(postRepository.findByTitle(title))
    }

    override fun findAllByArticle(article: Boolean): List<PostDto> {
        val articleList = postRepository.findAllByArticle(article)
        return articleList.map { PostDto.fromPost(it) }
    }
}